selfdrive/car/interfaces.py

```diff
@@ -518,3 +528,12 @@ def predict(self, x: list[float], do_sample: bool = False):
       pred = x[0]
     pred = pred * (self.weights['output_norm_mat'][1] - self.weights['output_norm_mat'][0]) + self.weights['output_norm_mat'][0]
     return pred
+
+
+SendCan = tuple[int, int, bytes, int]
+
+
+class CarControllerBase(CarInterfaceHelpers, ABC):
+  @abstractmethod
+  def update(self, CC, CS, now_nanos) -> tuple[car.CarControl.Actuators, list[SendCan]]:
+    pass
```

Comment:
Should be another PR, we can clean up more duplicate code with a base CC class, we just never got around to it

Reply:
ill just add the base class for now